docs/devel/hello-world-gadget.md

```diff
@@ -0,0 +1,522 @@
+	event.pid = bpf_get_current_pid_tgid() >> 32;
+
+	bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU,
+			      &event, sizeof(event));
+
+	return 0;
+}
+```
+
+Finally, it's needed to define the license of the eBPF code.
+
+```c
+char LICENSE[] SEC("license") = "GPL";
+```
+
+The full file should look like:
+
+```c
+// program.bpf.c
+
+// Kernel types definitions
+// Check https://blog.aquasec.com/vmlinux.h-ebpf-programs for more details
+#include <vmlinux.h>
+
+// eBPF helpers signatures
+// Check https://man7.org/linux/man-pages/man7/bpf-helpers.7.html to learn
+// more about different available helpers
+#include <bpf/bpf_helpers.h>
+
+// Inspektor Gadget macros
+#include <gadget/macros.h>
+
+struct event {
+	__u32 pid;
+};
+
+// we need to "use" this struct for the compiler not to remove it in an optimization pass
+const struct event *unusedevent __attribute__((unused));
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(u32));
+	__type(value, struct event);
+} events SEC(".maps");
+
+GADGET_TRACE_MAP(events);
+
+SEC("tracepoint/syscalls/sys_enter_openat")
+int enter_openat(struct trace_event_raw_sys_enter *ctx)
+{
+	struct event event = {};
+
+	event.pid = bpf_get_current_pid_tgid() >> 32;
+
+	bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU,
+			      &event, sizeof(event));
+
+	return 0;
+}
+
+char LICENSE[] SEC("license") = "GPL";
+```
+
+## Building the gadget for the first time
+
+We can now compile our gadget. You don't need to have any dependency on the machine, the `image
+build` by default uses a container with all dependencies to compile the code.
+
+```
+$ cd mygadget
+$ sudo -E ig image build -t mygadget:latest .
+INFO[0000] Experimental features enabled
+Successfully built docker.io/library/mygadget:latest@sha256:dd3f5c357983bb863ef86942e36f4c851933eec4b32ba65ee375acb1c514f628
+```
+
+## (Optional) Pushing the gadget image to a container registry
+
+You could push the gadget to a remote container registry. If you're using the same machine for
+building and running the gadget, this step can be skipped.
+
+```bash
+$ sudo -E ig image tag mygadget:latest ghcr.io/mauriciovasquezbernal/mygadget:latest
+INFO[0000] Experimental features enabled
+Successfully tagged with ghcr.io/mauriciovasquezbernal/mygadget:latest@sha256:dd3f5c357983bb863ef86942e36f4c851933eec4b32ba65ee375acb1c514f628
+
+$ sudo -E ig image push ghcr.io/mauriciovasquezbernal/mygadget:latest
+INFO[0000] Experimental features enabled
+Pushing ghcr.io/mauriciovasquezbernal/mygadget:latest...
+Successfully pushed ghcr.io/mauriciovasquezbernal/mygadget:latest@sha256:dd3f5c357983bb863ef86942e36f4c851933eec4b32ba65ee375acb1c514f628
+```
+## Running the gadget
+
+We're now all set to run our gadget for the first time.
+
+```bash
+$ sudo -E ig run mygadget:latest
+INFO[0000] Experimental features enabled
+RUNTIME.CONTAINERNAME                                                          PID
+WARN[0000] The gadget doesn't provide metadata
```

Comment:
Should the warning message include an hint to use `--update-metadata` (actionable error message)?

Reply:
This message is print at run time, there is nothing a user can do as `--update-metadata` is available at build time.